packages/framework/src/Component/Router/FriendlyUrl/FriendlyUrlGenerator.php

```diff
@@ -20,17 +22,41 @@ class FriendlyUrlGenerator extends BaseUrlGenerator
      */
     protected $friendlyUrlRepository;
 
+    /**
+     * @var \Shopsys\FrameworkBundle\Component\Router\FriendlyUrl\FriendlyUrlCacheKeyProvider|null
+     */
+    protected ?FriendlyUrlCacheKeyProvider $friendlyUrlCacheKeyProvider;
+
+    /**
+     * @var \Symfony\Contracts\Cache\CacheInterface|null
+     */
+    protected ?CacheInterface $mainFriendlyUrlSlugCache;
+
     /**
      * @param \Symfony\Component\Routing\RequestContext $context
      * @param \Shopsys\FrameworkBundle\Component\Router\FriendlyUrl\FriendlyUrlRepository $friendlyUrlRepository
+     * @param \Shopsys\FrameworkBundle\Component\Router\FriendlyUrl\FriendlyUrlCacheKeyProvider|null $friendlyUrlCacheKeyProvider
+     * @param \Symfony\Contracts\Cache\CacheInterface|null $mainFriendlyUrlSlugCache
      */
     public function __construct(
         RequestContext $context,
-        FriendlyUrlRepository $friendlyUrlRepository
+        FriendlyUrlRepository $friendlyUrlRepository,
+        ?FriendlyUrlCacheKeyProvider $friendlyUrlCacheKeyProvider = null,
+        ?CacheInterface $mainFriendlyUrlSlugCache = null
```

Comment:
This class is available as a service (thanks to autodiscovery for classes with `Generator` suffix) and also by the implemented setter injection via `setFriendlyUrlCacheKeyProvider`.

However, `$mainFriendlyUrlSlugCache` does not have the same cache pool if I request this class as a service and when used inside `FriendlyUrlFacade`. 

<details>
  <summary>Click to see dumped content</summary>
  
 
![Snímek obrazovky 2020-12-08 v 17 01 58](https://user-images.githubusercontent.com/1177414/101508027-7ca4d200-3977-11eb-8c3f-8a845adde852.png)


</details>

I suggest either remove setter injection (and maybe remove this class from autowiring in project base so it's not a bc break), or configure the service with proper cache pool)

Reply:
I have changed the name of cache pool.